File: quantum_vault.py
import uuid
import random
from abc import ABC, abstractmethod

class KuantumCokusuException(Exception):
    pass

class IKritik:
    def AcilDurumSogutmasi(self): ...

class KuantumNesnesi(ABC):
    def __init__(self, id_: str, stabilite_init: float, tehlike: int):
        self._id = id_
        self._stabilite = None
        self.stabilite = stabilite_init
        self.tehlike = tehlike

    @property
    def ID(self):
        return self._id

    @property
    def stabilite(self):
        return self._stabilite

    @stabilite.setter
    def stabilite(self, value):
        if value > 100:
            self._stabilite = 100
        elif value < 0:
            self._stabilite = 0
        else:
            self._stabilite = value

    @abstractmethod
    def AnalizEt(self):
        pass

    def DurumBilgisi(self):
        return f"{self.ID} - Stabilite: {self.stabilite:.2f}"

    def _check_collapse(self):
        if self.stabilite <= 0:
            raise KuantumCokusuException(f"Kuantum Çöküşü! Patlayan nesne: {self.ID}")

class VeriPaketi(KuantumNesnesi):
    def __init__(self, id_):
        super().__init__(id_, 100, 1)
    def AnalizEt(self):
        self.stabilite -= 5
        print("Veri içeriği okundu.")
        self._check_collapse()

class KaranlikMadde(KuantumNesnesi, IKritik):
    def __init__(self, id_):
        super().__init__(id_, 100, 8)
    def AnalizEt(self):
        self.stabilite -= 15
        print("Karanlık madde analiz edildi.")
        self._check_collapse()
    def AcilDurumSogutmasi(self):
        self.stabilite += 50
        if self.stabilite > 100: self.stabilite = 100
        print("Acil soğutma uygulandı.")

class AntiMadde(KuantumNesnesi, IKritik):
    def __init__(self, id_):
        super().__init__(id_, 100, 10)
    def AnalizEt(self):
        self.stabilite -= 25
        print("Evrenin dokusu titriyor...")
        self._check_collapse()
    def AcilDurumSogutmasi(self):
        self.stabilite += 50
        if self.stabilite > 100: self.stabilite = 100
        print("Acil soğutma uygulandı.")

def new_id(prefix):
    return f"{prefix}-{str(uuid.uuid4())[:8]}"

env = []

def find_by_id(id_):
    for e in env:
        if e.ID.lower() == id_.lower():
            return e
    return None

def main():
    try:
        while True:
            print("\nKUANTUM AMBARI KONTROL PANELİ")
            print("1. Yeni Nesne Ekle")
            print("2. Tüm Envanteri Listele")
            print("3. Nesneyi Analiz Et (ID ile)")
            print("4. Acil Durum Soğutması Yap (sadece kritik)")
            print("5. Çıkış")
            sec = input("Seçiminiz: ").strip()
            if sec == "1":
                t = random.randint(0,2)
                if t==0:
                    n = VeriPaketi(new_id("VP"))
                elif t==1:
                    n = KaranlikMadde(new_id("KM"))
                else:
                    n = AntiMadde(new_id("AM"))
                env.append(n)
                print("Yeni nesne eklendi:", n.DurumBilgisi())
            elif sec == "2":
                print("Envanter Durum Raporu:")
                for e in env:
                    print(e.DurumBilgisi())
            elif sec == "3":
                id_ = input("Analiz edilecek ID: ").strip()
                obj = find_by_id(id_)
                if not obj:
                    print("ID bulunamadı.")
                    continue
                obj.AnalizEt()
                print(obj.DurumBilgisi())
            elif sec == "4":
                id_ = input("Soğutulacak ID: ").strip()
                obj = find_by_id(id_)
                if not obj:
                    print("ID bulunamadı.")
                    continue
                if isinstance(obj, IKritik) or hasattr(obj, "AcilDurumSogutmasi"):
                    obj.AcilDurumSogutmasi()
                    print(obj.DurumBilgisi())
                else:
                    print("Bu nesne soğutulamaz!")
            elif sec == "5":
                print("Çıkılıyor...")
                break
            else:
                print("Geçersiz seçim.")
    except KuantumCokusuException as ex:
        print("\nSİSTEM ÇÖKTÜ! TAHLİYE BAŞLATILIYOR...")
        print(ex)
if __name__ == "__main__":
    main()

File: quantum_vault.js
const readline = require('readline');
const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

class KuantumCokusuException extends Error {
    constructor(id) { super(`Kuantum Çöküşü! Patlayan nesne: ${id}`); }
}

class KuantumNesnesi {
    constructor(id, stabiliteInit, tehlike) {
        this._id = id;
        this._stabilite = null;
        this.stabilite = stabiliteInit;
        this.tehlike = tehlike;
    }
    get ID() { return this._id; }
    get stabilite() { return this._stabilite; }
    set stabilite(v) {
        if (v > 100) this._stabilite = 100;
        else if (v < 0) this._stabilite = 0;
        else this._stabilite = v;
    }
    AnalizEt() { throw new Error("Abstract"); }
    DurumBilgisi() { return `${this.ID} - Stabilite: ${this.stabilite.toFixed(2)}`; }
    _checkCollapse() {
        if (this.stabilite <= 0) throw new KuantumCokusuException(this.ID);
    }
}

class VeriPaketi extends KuantumNesnesi {
    constructor(id) { super(id, 100, 1); }
    AnalizEt() {
        this.stabilite -= 5;
        console.log("Veri içeriği okundu.");
        this._checkCollapse();
    }
}

class KaranlikMadde extends KuantumNesnesi {
    constructor(id) { super(id, 100, 8); }
    AnalizEt() {
        this.stabilite -= 15;
        console.log("Karanlık madde analiz edildi.");
        this._checkCollapse();
    }
    AcilDurumSogutmasi() {
        this.stabilite += 50;
        if (this.stabilite > 100) this.stabilite = 100;
        console.log("Acil soğutma uygulandı.");
    }
}

class AntiMadde extends KuantumNesnesi {
    constructor(id) { super(id, 100, 10); }
    AnalizEt() {
        this.stabilite -= 25;
        console.log("Evrenin dokusu titriyor...");
        this._checkCollapse();
    }
    AcilDurumSogutmasi() {
        this.stabilite += 50;
        if (this.stabilite > 100) this.stabilite = 100;
        console.log("Acil soğutma uygulandı.");
    }
}

const env = [];
const rnd = { next: (n)=> Math.floor(Math.random()*n) };
const newId = (prefix) => `${prefix}-${Math.random().toString(36).slice(2,10)}`;

function prompt(q) {
    return new Promise(res => rl.question(q, ans => res(ans)));
}

async function main() {
    try {
        while (true) {
            console.log("\nKUANTUM AMBARI KONTROL PANELİ");
            console.log("1. Yeni Nesne Ekle");
            console.log("2. Tüm Envanteri Listele");
            console.log("3. Nesneyi Analiz Et (ID ile)");
            console.log("4. Acil Durum Soğutması Yap (sadece kritik)");
            console.log("5. Çıkış");
            let sec = (await prompt("Seçiminiz: ")).trim();
            if (sec === "1") {
                let t = rnd.next(3);
                let n;
                if (t===0) n = new VeriPaketi(newId("VP"));
                else if (t===1) n = new KaranlikMadde(newId("KM"));
                else n = new AntiMadde(newId("AM"));
                env.push(n);
                console.log("Yeni nesne eklendi:", n.DurumBilgisi());
            } else if (sec === "2") {
                console.log("Envanter Durum Raporu:");
                env.forEach(e => console.log(e.DurumBilgisi()));
            } else if (sec === "3") {
                let id = (await prompt("Analiz edilecek ID: ")).trim();
                let found = env.find(x => x.ID.toLowerCase() === id.toLowerCase());
                if (!found) { console.log("ID bulunamadı."); continue; }
                found.AnalizEt();
                console.log(found.DurumBilgisi());
            } else if (sec === "4") {
                let id = (await prompt("Soğutulacak ID: ")).trim();
                let found = env.find(x => x.ID.toLowerCase() === id.toLowerCase());
                if (!found) { console.log("ID bulunamadı."); continue; }
                if (typeof found.AcilDurumSogutmasi === 'function') {
                    found.AcilDurumSogutmasi();
                    console.log(found.DurumBilgisi());
                } else {
                    console.log("Bu nesne soğutulamaz!");
                }
            } else if (sec === "5") {
                console.log("Çıkılıyor...");
                break;
            } else console.log("Geçersiz seçim.");
        }
    } catch (ex) {
        if (ex instanceof KuantumCokusuException) {
            console.log("\nSİSTEM ÇÖKTÜ! TAHLİYE BAŞLATILIYOR...");
            console.log(ex.message);
        } else {
            console.log("Beklenmeyen hata:", ex);
        }
    } finally {
        rl.close();
    }
}

main();
